src/parser/expressions_parser.rs

```diff
@@ -305,10 +317,12 @@ fn parse_array_literal(lexer: &mut ParseSession) -> Result<AstNode, Diagnostic>
     let start = lexer.range().start;
     lexer.expect(KeywordSquareParensOpen)?;
     lexer.advance();
+    lexer.paren = true;
     let elements = Some(Box::new(parse_expression(lexer)));
```

Comment:
everything parsed in here will preseve the ParenthesizedExpressions. Can this become a problem?

```st
ARRAY[1..2] OF foo := [(idx := (0), val := (((0)))), (idx := (1+3), val := 1)];
```

is there a way I can sneak a ParensExpressions into codegen?